nft/invoice_unpaid.go

```diff
@@ -82,13 +82,17 @@ type MintNFTRequest struct {
 // InvoiceUnpaid handles transactions related to minting of NFTs for unpaid invoices
 type InvoiceUnpaid interface {
 	// MintNFT mints an NFT
-	MintNFT(ctx context.Context, request MintNFTRequest) (*MintNFTResponse, chan bool, error)
+	MintNFT(ctx context.Context, request MintNFTRequest) (*Response, chan bool, error)
 	// GetRequiredInvoiceUnpaidProofFields returns the required proof field properties
 	GetRequiredInvoiceUnpaidProofFields(ctx context.Context) ([]string, error)
+	// TransferFrom transfers an NFT to another address
+	TransferFrom(ctx context.Context, registry common.Address, to common.Address, tokenID TokenID) (*Response, chan bool, error)
+	// OwnerOf returns the owner of an NFT
+	OwnerOf(registry common.Address, tokenID []byte) (owner common.Address, err error)
 }
 
-// MintNFTResponse holds tokenID and transaction ID.
-type MintNFTResponse struct {
+// Response holds tokenID and transaction ID.
```

Comment:
what was the reasoning behind making this naming more generic?

Reply:
Because I could reuse the same struct for `mint` and `transfer`.

Both should return the `jobID` and the `tokenID`

Reply:
IMO, TokenResponse might be specific instead of Response which seems too generic. Also, do we need to export this structure?